moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -1310,4 +1310,33 @@ double TimeOptimalTrajectoryGeneration::verifyScalingFactor(const double request
   }
   return scaling_factor;
 }
+
+trajectory_msgs::msg::JointTrajectory createTrajectoryMessage(const std::vector<std::string>& joint_names,
+                                                              const trajectory_processing::Trajectory& trajectory,
+                                                              const int sampling_rate)
+{
+  trajectory_msgs::msg::JointTrajectory trajectory_msg;
+  trajectory_msg.joint_names = joint_names;
+  assert(sampling_rate > 0);
```

Comment:
Should we add an error message here in case the condition fails here?

Reply:
I think so, since this is a public function that anyone can call, we should check for invalid inputs and return a proper error.
In Release mode the assert wouldn't run and we could have undefined behavior below because of a division by zero.